src/PensionsRegulatorApi/Application/Queries/GetOrganisationsHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using MediatR;
+using PensionsRegulatorApi.Data;
+using PensionsRegulatorApi.Domain;
+
+namespace PensionsRegulatorApi.Application.Queries
+{
+    
+
+    public class GetOrganisationsHandler : RequestHandler<GetOrganisations, IEnumerable<Organisation>>
+    {
+        private readonly OrganisationRepository _repository;
+
+        public GetOrganisationsHandler(OrganisationRepository repository)
+        {
+            _repository = repository ?? throw new ArgumentNullException(nameof(repository));
+        }
+        
+        protected override IEnumerable<Organisation> Handle(GetOrganisations request)
+        {
+            try
+            {
+                return
+                    _repository
+                        .GetOrganisationsForPAYEReference(
+                            request.PAYEReference);
+            }
+            catch
+            {
+                return 
+                    Enumerable
+                        .Empty<Organisation>();
```

Comment:
Is returning an empty set the right thing here? If an exception has thrown should we log it? I suppose that if there is an exception we would want to return an empty set to the UI, but feels wrong to do that at the handler level. I have added logging into the api so I can add logging to this when I do the endpoint if required.

Reply:
@tom-gough I wasn't sure about this, that's why I called it out at top of review. I can let exception bubble up to controller and log there np.

Alternative is to wrap up Handler results in something like `Result<T>`, but I haven't seen that pattern used around here.

Reply:
Changed to propagate the exception up.